lib/pbench/test/unit/server/conftest.py

```diff
@@ -69,6 +75,9 @@
 admin_email = "test_admin@example.com"
 generic_password = "12345"
 jwt_secret = "my_precious"
+# oidc public client name needs to match with the one coming
+# from the config file
```

Comment:
Ambiguous reference? We're creating a Pbench "config file" here, and while you've made that match with the *default* value (which this overrides anyway), that doesn't actually need to match. And we're not running against a real Keycloak here, so what *does* it need to match? I searched briefly for hardcoded reference and didn't find one; so if this comment is really important, a bit more context would be helpful.

Reply:
Okay so since we are exposing `openid-connect` section in the config file, Server is going to use OIDC-based authentication i.e. `verify_auth_oidc` from the auth.py file to decode the tokens. This function uses `token_introspect` internally to decode the token locally (offline). `token_introspect` function uses the `jwt` to decode the token without contacting the Keycloak but it also needs to verify the `aud` claim of the token.

In the server code, the `audience` value is the `client_id` that we get from the server config file. Therefore if the encoded token has `aud` claim anything other than this `client_id` it would be rejected by the verify auth function. Therefore when we encode the token we have to encode the correct `client_id` as the audience claim in the token, look at the conftest.py `genrate_token` function line 868. Thats why the comment said that this value has to match with the config file client id.

Reply:
Perhaps the issue is that we're _not_ creating a Pbench "config file" **here** -- that creation ended at line 72.  What we are doing here is defining module-level Python global variables...hence the desire for a big, fat code comment.

Perhaps, we should be reading the value from the config file, instead....

Reply:
Yeah perhaps reading the value from the config file is better, but that means I have to pass the server_config fixture to all the fixtures trying to generate the token like `pbench_drb_token/pbench_admin_token` etc. I guess it's not too bad we pass around fixtures pretty casually.

Reply:
I don't know what the rules are, but perhaps you could try referencing the fixture _here_.  (If the fixture is in scope and has a fixture-scope of `module` or `session`, it would probably be "alive"....)

Or, you could just create a fixture for the client name and include it where it's needed....

Reply:
> could try referencing the fixture _here_

But that means I have to make the generate_token function a fixture.

> Or, you could just create a fixture for the client name and include it where it's needed....

That still means passing that fixture everywhere so I might just use the already defined `server_config` fixture.

Reply:
I think the line that we're kvetching about looks like this:
```
oidc_public_client = "pbench-dashboard"
```
and the challenge that we're trying to confront is where to get the `"pbench-dashboard"` value from.  And, one possibility is to get it from the `server_config()` fixture (which _does_ have a `session` "scope"), _iff_ we can call it here at module scope (i.e., where we are not inside a test function and/or another fixture).

Another option (if we can tolerate this level of hackery) would be initialize it to `None` here and have the `server_config()` fixture set it (referencing it as a `global`).

The point is to avoid having to pass it everywhere (which is, presumably, why you declared it to be a "global variable" in the first place), and the question is simply how to get it initialized appropriately.